### PR TITLE
[INTEL MKL] Slice and Reshape op support with MKLDNN 1.0

### PR DESCRIPTION
**Pull request by mdfaijul**
This PR updates kernels for SliceOp and ReshapeOp to enable MKLDNN 1.0 functionality.

### CODE REVIEW

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/core/kernels/mkl_slice_op.cc [hunk 1]
```diff
@@ -187,7 +191,17 @@ class MklSlicePrimitive : public MklPrimitive {
   void Execute(const MklSliceParams& sliceParams) {
     context_.src_mem->set_data_handle(sliceParams.from->get_data_handle());
     context_.dst_mem->set_data_handle(sliceParams.to->get_data_handle());
+
+#ifdef ENABLE_MKLDNN_V1
```
[the remark this answers is not in the document]
Comment (nammbash) [reply tensorflow/core/kernels/mkl_slice_op.cc]: @penpornk That specific change is not addressable in near term. So No refactor is attempted. See the PR #36496  for comments.

[the remark this answers is not in the document]
Comment (nammbash) [reply tensorflow/core/kernels/mkl_slice_op.cc]: @penpornk @mdfaijul Changes Done in the other PR. FYI

[the remark this answers is not in the document]
Comment (mdfaijul) [reply tensorflow/core/kernels/mkl_slice_op.cc]: @penpornk Addressed the comments.

**Comment by mdfaijul** [comment]
@penpornk Thanks a lot for reviewing this PR. I have addressed all your comments. Please check.

File: tensorflow/core/util/mkl_util.h [hunk 2]
```diff
@@ -728,9 +732,9 @@ inline Status ConvertMklToTF(OpKernelContext* context,
     }
     return Status::OK();
   } catch (mkldnn::error& e) {
-    string error_msg = "Status: " + std::to_string(e.status) +
-                       ", message: " + string(e.message) + ", in file " +
-                       string(__FILE__) + ":" + std::to_string(__LINE__);
+    string error_msg = "Status: " + std::to_string(e.status) + ", message: " +
+                       string(e.message) + ", in file " + string(__FILE__) +
+                       ":" + std::to_string(__LINE__);
```
Comment (penpornk) [inline tensorflow/core/util/mkl_util.h]: The formatting here should remain the same (before change). I'll email you the `clang` version.

File: tensorflow/core/util/mkl_util.h [hunk 3]
```diff
@@ -1250,8 +1254,8 @@ inline Status CreateBlockedMemDescHelper(const memory::dims& dim,
   } catch (mkldnn::error& e) {
     return Status(error::Code::INTERNAL,
                   tensorflow::strings::StrCat(
-                      "Failed to create blocked memory descriptor.",
-                      "Status: ", e.status, ", message: ", e.message));
+                      "Failed to create blocked memory descriptor.", "Status: ",
+                      e.status, ", message: ", e.message));
```
Comment (penpornk) [inline tensorflow/core/util/mkl_util.h]: The formatting here should remain the same too.

File: tensorflow/core/kernels/mkl_slice_op.cc [hunk 4]
```diff
@@ -193,12 +192,8 @@ class MklSlicePrimitive : public MklPrimitive {
...
 
 #ifdef ENABLE_MKLDNN_V1
-    DCHECK_EQ(context_.slice_primitives.size(),
-              context_.slice_primitives_args.size());
-    for (size_t i = 0; i < context_.slice_primitives.size(); ++i) {
-      context_.slice_primitives.at(i).execute(
-          *context_.slice_stream, context_.slice_primitives_args.at(i));
-    }
+    execute_primitives(context_.slice_primitives, context_.slice_stream,
+                      context_.slice_primitives_args);
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_slice_op.cc]: This looks like it needs one more space. Please apply `clang-format`.
```suggestion
                       context_.slice_primitives_args);
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the changes!

File: tensorflow/core/kernels/mkl_slice_op.cc (hunk 4 above)
Comment (mdfaijul) [reply tensorflow/core/kernels/mkl_slice_op.cc]: @penpornk Applied clang-format 8.0.1. Please check.
Thanks!

**Review by penpornk — APPROVED** [review]
Thank you very much!

**Pull request merged**